File: src/main/java/br/com/poketeams/model/service/CoachService.java
package br.com.poketeams.model.service;

import br.com.poketeams.model.entity.Coach;

import java.util.List;

public interface CoachService {

    Coach create(Coach coach);

    Coach findOne(Long coachId);

    List<Coach> findAll();

    Coach update(Coach coach);

    Coach delete(Long coachId);
}

File: src/main/resources/static/js/service/coach-service.js
app.factory('coachService', function ($http) {

    function create(coach) {
        return $http.post('api/coach', coach);
    }

    function findOne(coachId) {
        return $http.get('api/coach/' + coachId);
    }

    function findAll() {
        return $http.get('api/coach');
    }

    function update(coach) {
        return;
    }

    function deleteOne(coachId) {
        return;
    }

    return {
        create: create,
        findOne: findOne,
        findAll: findAll,
        update: update,
        deleteOne: deleteOne
    }

});